Rollin' Into Main

jeffaburke opened the pull request:
Added A Default Drive Map
Added Drive functionality
Changed Swerve Module for General Use
Added Mock Swerve Module
Added Robot Controls
Added Gaston Map

jeffaburke committed:
Added Drive File

jeffaburke committed:
Discarded Unnecessary Function

jeffaburke committed:
Matt's bug

jeffaburke committed:
We have a mock swerve module

Takes in a location

jeffaburke committed:
Added a default drive map

jeffaburke committed:
Added a Drive Constructor

Sets all private variables

jeffaburke committed:
Added Drive into Robot

jeffaburke committed:
Adding Necessary functionality for swerve

jeffaburke committed:
Adding Drive Functionality

jeffaburke committed:
Starting to add button bindings

jeffaburke committed:
Added Joystick Drive Control

jeffaburke committed:
Cleaning Up some code and adding gyro reset

jeffaburke committed:
Gaston Map

msoucy commented on src/main/java/com/chopshop166/chopshoplib/drive/MockSwerveModule.java:
```diff
@@ -0,0 +1,60 @@
+
+public class MockSwerveModule implements SwerveModule {
+
+    private Translation2d location = new Translation2d();
```
Why is this getting a default value? Just set it from the constructor. Also make it final.

jeffaburke replied:
I I made this final and now it is declared in the constructor

msoucy commented on src/main/java/frc/robot/maps/GastonMap.java:
```diff
@@ -0,0 +1,73 @@
+                new PIDSparkMax(2, MotorType.kBrushless), new PIDSparkMax(1, MotorType.kBrushless),
+                SDSSwerveModule.MK4_V2); // TODO Change controller ids
+
+        // final Translation2d moduleLocation, final CANCoder steeringEncoder,
```
Commented out code

jeffaburke replied:
Took this out

msoucy commented on src/main/java/frc/robot/maps/RobotMap.java:
```diff
@@ -1,9 +1,93 @@
+
+        public DriveMap() {
+
+            this.frontLeft = new MockSwerveModule(new Translation2d(0.381, 0.381));
```
Maybe make `0.381` a constant? where did this number come from? What unit is it in? (Yes they're mocks, but it's still good practice)

bot190 commented on src/main/java/frc/robot/maps/GastonMap.java:
```diff
@@ -0,0 +1,73 @@
+        final CANCoder encoderFR = new CANCoder(2); // TODO Get cancoder id
+        encoderFR.configMagnetOffset(0); // TODO Get Magnet Offset
+        encoderFR.configAbsoluteSensorRange(AbsoluteSensorRange.Unsigned_0_to_360);
+        final SDSSwerveModule frontRight = new SDSSwerveModule(new Translation2d(MODULE_OFFSET_XY, MODULE_OFFSET_XY),
```
The robot is going to be symmetrical so we can use the same offset. But some of these values need to be negative to represent the correct locations.

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -0,0 +1,109 @@
+    return running("Field Centric Drive", () -> handleSwerve(translateX, translateY, rotation));
+  }
+
+  private void handleSwerve(final DoubleSupplier translateX, final DoubleSupplier translateY,
```
`handleSwerve`? What about swerve does it handle?

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -0,0 +1,109 @@
+
+  @Override
+  public void safeState() {
+    // TODO Stop all movement
```
Please implement this

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -0,0 +1,109 @@
+    return functional("Drive Distance Y", () -> {
+      frontLeft.resetDistance();
+    }, () -> {
+      handleSwerve(() -> 0, () -> Math.signum(distance) * 0.2, () -> 0);
```
Where does `0.2` come from?

jeffaburke replied:
tbh idk

bot190 commented on src/main/java/frc/robot/maps/GastonMap.java:
```diff
@@ -0,0 +1,73 @@
+
+        // All Distances are in Meters
+        // Front Left Module
+        final CANCoder encoderFL = new CANCoder(1); // TODO Get cancoder id
```
We can configure the `CANCoder` to match these ids. No need to make these a `TODO`

bot190 commented on src/main/java/frc/robot/maps/GastonMap.java:
```diff
@@ -0,0 +1,73 @@
+        final SDSSwerveModule frontLeft = new SDSSwerveModule(new Translation2d(MODULE_OFFSET_XY, MODULE_OFFSET_XY),
+                encoderFL,
+                new PIDSparkMax(2, MotorType.kBrushless), new PIDSparkMax(1, MotorType.kBrushless),
+                SDSSwerveModule.MK4_V2); // TODO Change controller ids
```
Same with the controller ids.

bot190 commented on src/main/java/frc/robot/maps/RobotMap.java:
```diff
@@ -1,9 +1,93 @@
 // Need to get MAC address for roborio
 @RobotMapFor("Default")
 public class RobotMap {
+    public static class DriveMap {
```
We should pull this into chopshoplib at the end of the season alongside the swerve module code.

msoucy replied:
Would you mind calling this `SwerveDriveMap`?

bot190 commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -0,0 +1,109 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
```
Please remove the wpilib copyright notice.

msoucy commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -0,0 +1,109 @@
+// Copyright (c) FIRST and other WPILib contributors.
```
Please remove this copyright header

jeffaburke committed:
Start of PR Comments

jeffaburke committed:
Pull Request Update

Removed WPI Copyright
Changed `handleSwerve` to `updateSwerveSpeedAngle` for clarity
Removed useless TODO's
Added `safeState` stopping functionality
Changed Module Offset to make them to a closer accuracy

msoucy commented on src/main/java/frc/robot/maps/GastonMap.java:
```diff
@@ -0,0 +1,73 @@
+        encoderFL.configMagnetOffset(0); // TODO Get Magnet Offset
+        encoderFL.configAbsoluteSensorRange(AbsoluteSensorRange.Unsigned_0_to_360);
+        final SDSSwerveModule frontLeft = new SDSSwerveModule(
+                new Translation2d(MODULE_OFFSET_XY, MODULE_OFFSET_XY),
```
The translations are definitely wrong (0-value aside)

jeffaburke replied:
why is this? i took them from 2022-swerve

jeffaburke replied:
disregard my last comment

msoucy commented on src/main/java/frc/robot/maps/GastonMap.java:
```diff
@@ -0,0 +1,73 @@
+
+        // Value taken from CAD as offset from center of module base pulley to center of
+        // robot
+        final double MODULE_OFFSET_XY = 0; // TODO Get this value
```
Default the offset to 1 just so we hopefully avoid straight-up crashes. Or even better, if you can get the real size from CAD that would be great.

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -23,7 +33,7 @@ public void populateDashboard() {
   public void setDefaultCommands() {
-    // TODO Auto-generated method stub
-
+    drive.setDefaultCommand(drive.fieldCentricDrive(() -> driveController.getLeftX(),
```
Pro tip: you can use `driveController::getLeftX` instead of `() -> driveController.getLeftX()`

jeffaburke replied:
why is this?

msoucy replied:
Just a little shorter - you're taking a reference to the function itself, instead of making a lambda that calls it. (Theoretically it might be a little faster, but I wouldn't be bothered about that)

jeffaburke committed:
Controller and Swerve Location Fix

Made the swerve modules correct locations,
Simplified the controller code

jeffaburke committed:
Rename Drive to Swerve Drive

jeffaburke committed:
Format

jeffaburke committed:
Merge branch 'main' into drive

jeffaburke committed:
Syntax fixing after merge

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -25,23 +29,24 @@ public void robotInit() {
 
-    controller.start().whenPressed(parallel("Stop", leftClimber.stop(), rightClimber.stop()));
-
+    driveController.start().whenPressed(parallel("Stop", leftClimber.stop(), rightClimber.stop()));
```
Just a warning, this is also mapped to `drive.resetCmd()` up above.

bot190 replied:
Yeah probably good to remap this to `select()` instead (and maybe do the same for all commands?)

bot190 commented on src/main/java/frc/robot/maps/GastonMap.java:
```diff
@@ -1,27 +1,91 @@
+        public SwerveDriveMap getSwerveDriveMap() {
+                // Value taken from CAD as offset from center of module base pulley to center of
+                // robot
+                final double MODULE_OFFSET_XY = 0.314325; // TODO Get this value
```
Isn't this "TODONE"?

bot190 reviewed (commented):
2 Minor comments. So close.

jeffaburke committed:
Merge branch 'main' into drive

jeffaburke committed:
Fixes for PR
